File: flutter_inappwebview_forge_android/android/src/main/kotlin/com/emirkanacar/flutter_inappwebview_forge_android/download_job/DownloadJobChannelDelegate.kt
package com.emirkanacar.flutter_inappwebview_forge_android.download_job

import com.emirkanacar.flutter_inappwebview_forge_android.types.ChannelDelegateImpl
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

open class DownloadJobChannelDelegate(
    private var downloadJobController: DownloadJobController?,
    channel: MethodChannel
) : ChannelDelegateImpl(channel) {

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "cancel" -> {
                downloadJobController?.cancel()
                result.success(true)
            }
            "getInfo" -> result.success(downloadJobController?.getInfo())
            "dispose" -> {
                downloadJobController?.dispose()
                result.success(true)
            }
            else -> result.notImplemented()
        }
    }

    fun onProgressChanged(progress: Double) {
        getChannel()?.invokeMethod(
            "onProgressChanged",
            hashMapOf("progress" to progress)
        )
    }

    fun onComplete(completed: Boolean, error: String?) {
        getChannel()?.invokeMethod(
            "onComplete",
            hashMapOf("completed" to completed, "error" to error)
        )
    }

    override fun dispose() {
        super.dispose()
        downloadJobController = null
    }
}

File: flutter_inappwebview_forge_ios/ios/flutter_inappwebview_forge_ios/Sources/flutter_inappwebview_forge_ios/DownloadJob/DownloadJobChannelDelegate.swift
import Foundation
import Flutter

public class DownloadJobChannelDelegate: ChannelDelegate {
    private weak var downloadJobController: DownloadJobController?

    public init(downloadJobController: DownloadJobController, channel: FlutterMethodChannel) {
        super.init(channel: channel)
        self.downloadJobController = downloadJobController
    }

    public override func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "cancel":
            downloadJobController?.cancel()
            result(true)
        case "getInfo":
            result(downloadJobController?.getInfo())
        case "dispose":
            downloadJobController?.dispose()
            result(true)
        default:
            result(FlutterMethodNotImplemented)
        }
    }

    public func onProgressChanged(progress: Double) {
        channel?.invokeMethod("onProgressChanged", arguments: ["progress": progress])
    }

    public func onComplete(completed: Bool, error: String?) {
        channel?.invokeMethod("onComplete", arguments: ["completed": completed, "error": error])
    }

    public override func dispose() {
        super.dispose()
        downloadJobController = nil
    }
}
